DOCS: Updates docs to reflect Launcher changes

cadihowley opened the pull request:
- updates screenshots to show Launcher grid view/changed settings tabs
- updates all settings pages to reflect new Launcher button terminology
- moves Engine tab content to Game tab page
- removes Engine tab from index
- fixes some broken refs/other small errors


<!---
Thank you for contributing to ScummVM. Please read the following carefully before submitting your Pull Request.

Make sure your individual commits follow the guidelines found in the ScummVM Wiki: https://wiki.scummvm.org/index.php?title=Commit_Guidelines. If they're not please edit them before submitting the Pull Request.

Proper documentation must also be included for common code and changes impacting user facing elements.

Commits and Pull Requests should use the following template:

```
SUBSYSTEM: Short (50 chars or less) summary of changes

More detailed explanatory text, if necessary.  Wrap it to about 72
characters or so.  In some contexts, the first line is treated as the
subject of an email and the rest of the text as the body.  The blank
line separating the summary from the body is critical (unless you omit
the body entirely); tools like rebase can get confused if you run the
two together.

Write your commit message in the present tense: "Fix bug" and not "Fixed
bug."  This convention matches up with commit messages generated by
commands like git merge and git revert.

Further paragraphs come after blank lines.

- Bullet points are okay, too

- Typically a hyphen or asterisk is used for the bullet, preceded by a
 single space, with blank lines in between, but conventions vary here

- Use a hanging indent
```
--->

cadihowley commented:
Let me know if there's anything missing - or if there's anything else I should add while I'm at it!

criezy reviewed (approved):
The changes look good to me.

The new images look a bit blurry to me, especially when viewed on a non-Retina screen, but that was the case for most of the old images as well. I am not sure the change of format (using jpeg instead of png) helps though.

cadihowley commented:
> The new images look a bit blurry to me, especially when viewed on a non-Retina screen, but that was the case for most of the old images as well. I am not sure the change of format (using jpeg instead of png) helps though.

Hmm I have to agree with you. Had some issues getting the latest build installed on my Linux machine so the screenshots are coming off my Android tablet, hence the jpg.... but as mentioned in Discord I think I'll be redoing ALL the screenshots on a mac anyway! Cheers for the feedback

Thunderforge commented:
It looks like the launcher screenshots all have `2.7.0git` as their version.

If these are going to be the official docs for 2.6.0, they probably should say `2.6.0` instead (compare with `2.2.0` in many of the ones being replaced). I think it looks more professional that way as well since we're not using a daily build.

All that said, I don't know that I feel strongly enough about it to hold up the PR, especially if the screenshots are going to be redone in the near future to better accommodate retina screens.

Thunderforge commented on doc/docportal/settings/cloud.rst:
```diff
@@ -6,12 +6,12 @@ Use the Cloud tab to manage connected cloud storage services.
 For a comprehensive look at how to use ScummVM's built-in cloud functionality, check out our :doc:`../use_scummvm/connect_cloud` guide.
 
-From the Launcher, select **Options**, click the **>** scroll arrow until the Cloud tab is visible, and then select the **Cloud** tab.
+From the Launcher, select **Global Options**, if necessery click the **>** scroll arrow until the Cloud tab is visible, and then select the **Cloud** tab.
```
Typo, should be `necessary`

cadihowley replied:
> If these are going to be the official docs for 2.6.0, they probably should say `2.6.0` instead (compare with `2.2.0` in many of the ones being replaced). I think it looks more professional that way as well since we're not using a daily build.

I wasn't really intending these to be the docs for 2.6.0 necessarily - if they're going to be then I'll need a hand to create a pull request against branch-2-6. Not sure how to do that without making a mess of things! 

With regards to the screenshots, up until now (disregarding the initial v2.2 docs), screenshots have not been updated with each release. In an ideal world, any PR making changes to the GUI would have accompanying new screenshots and documentation. This would result in docs and screenshots that are always up to date in terms of functionality, without the screenshot launcher versions necessarily matching - but as you say, this does look less professional. The alternative is that prior to each release I create an entirely new set of screenshots, which would probably look best, but is obviously the most labour intensive option!

SupSuper replied:
> I wasn't really intending these to be the docs for 2.6.0 necessarily - if they're going to be then I'll need a hand to create a pull request against branch-2-6. Not sure how to do that without making a mess of things!

We can just cherry-pick this to branch-2-6 after it's merged.

> With regards to the screenshots, up until now (disregarding the initial v2.2 docs), screenshots have not been updated with each release. In an ideal world, any PR making changes to the GUI would have accompanying new screenshots and documentation. This would result in docs and screenshots that are always up to date in terms of functionality, without the screenshot launcher versions necessarily matching - but as you say, this does look less professional. The alternative is that prior to each release I create an entirely new set of screenshots, which would probably look best, but is obviously the most labour intensive option!

Maybe we can just hide the version number from the screenshots?

cadihowley replied:
>Maybe we can just hide the version number from the screenshots?

Maybe just blurring the version number would work?
![launcher_list_group](https://user-images.githubusercontent.com/64062738/177108253-0e8508e5-71c3-4c17-bcf7-301d1743993d.png)

Thunderforge commented on doc/docportal/use_scummvm/taskbar_integration.rst:
```diff
@@ -9,14 +9,13 @@ The taskbar integration has several features:
 Icons for many games can be found on our `Github repository <https://github.com/scummvm/scummvm-icons>`__.
 
-Place the icons in a folder, and either change the *iconspath* in the :doc:`configuration file <../advanced_topics/configuration_file>` to point to this folder, or add the folder to the :ref:`Extra Path <extra>` on the Paths settings tab.
+Place the icons in a folder, and change either the :ref:`Icon Path <iconspath>` or the :ref:`Extra Path <extra>` to point to this folder. 
```
FYI, #4024 will likely be backported to 2.6.0. This would change the instructions.

cadihowley replied:
Thanks, I'll have a look and update accordingly!

Thunderforge commented:
GitHub reviews don't seem to let me add review comments to images, so here are a couple of things:

 * `Launcher/launcher_popup.jpg` has markings in the upper right for input type, which I think is an Android-exclusive thing. There's also a `launcher_popup.png` and I'm not sure if there should be both.
 * `settings/audio_game.png` has the audio override checkbox enabled in the old, but disabled in the new. It's a bit hard to cross-reference on GitHub with what the documentation is saying, but I want to make sure it's showing the right thing.
 * `settings/cloud.jpg` shows far less info in the new version than the old. Not sure if that's intentional.

Thunderforge commented on doc/docportal/settings/game.rst:
```diff
@@ -41,4 +41,669 @@ All settings can also be changed in the :doc:`../advanced_topics/configuration_f
+.. _sitcom:
+
+Sitcom mode
+	Adds laughter after the actor's line or narration.
```
🤣

sev- reviewed (commented):
Very nice change. Speaking of 2.7.0git screenshots, what you can do, @cadihowley is use nightly build. Only the initial launcher screenshot is missing.

What I would also suggest to add is a sample with groups in the launcher list (not the icons view but the list view).

Also, it would be awesome to have couple of screenshots for "Update icons" dialog. (Options -> Misc, there is a button, you click it and a dialog appears).

cadihowley commented:
>     * `Launcher/launcher_popup.jpg` has markings in the upper right for input type, which I think is an Android-exclusive thing. There's also a `launcher_popup.png` and I'm not sure if there should be both.

No, there shouldn't be! I thought I removed all the .jpg files on the second commit, but I was in a bit of a time crunch before going to work so I clearly missed one or two! 
  
>     * `settings/audio_game.png` has the audio override checkbox enabled in the old, but disabled in the new. It's a bit hard to cross-reference on GitHub with what the documentation is saying, but I want to make sure it's showing the right thing.

Good pickup - thanks 

>     * `settings/cloud.jpg` shows far less info in the new version than the old. Not sure if that's intentional.

Not entirely - the .jpg file is the wrong one, but the .png is also not the one that should be in there. I was drowning in screenshots and renamed the wrong one I think 🤣

cadihowley commented:
> What I would also suggest to add is a sample with groups in the launcher list (not the icons view but the list view).
>
> Also, it would be awesome to have couple of screenshots for "Update icons" dialog. (Options -> Misc, there is a button, you click it and a dialog appears).

@sev- Consider it done!

cadihowley commented:
@Thunderforge @sev- 
If you're happy with these commits for the 2.6 docs, please merge/cherry pick as required for branch-2-6. Sorry that it's last minute but unfortunately I won't be able to do any more before the branch is frozen!

criezy commented:
Thank you for the updated pictures. We discussed a bit on Discord yesterday and decided to treat the documentation as a special case, and it will not be subject to the freeze this evening. That leaves us a few more days and I will have time on Sunday to look again at this pull request and make adjustments if needed before merging and backporting to the release branch.

criezy commented on doc/docportal/use_scummvm/taskbar_integration.rst:
```diff
@@ -4,18 +4,19 @@ Taskbar integration
 ================================
 
-Icons for many games can be found on our `Github repository <https://github.com/scummvm/scummvm-icons>`__.
+To download or update game icons, from the Launcher go to **Global Options**, go to the **Misc** tab, and click on the **Update Icons** button. See the :ref:`Update Icons <updateicons>` for more information.
```
I will modify this when merging as currently the taskbar is unable to use the icons.dat file that we can download from the Global Options. So icons still need to be downloaded manually.

cadihowley replied:
Ahh thank you - I should've double checked this one, I just assumed!

criezy reviewed (approved):
That looks good to me and I think all the requested changes have been done. I noticed one issue with the taskbar doc update, but I will merge the PR manually and do the change myself during the merge.

criezy commented:
I have now merged the changes manually and done a few additional changes:
 - Blurred the version in one additional image
 - Squashed a few commits together
 - Updated the documentation for the taskbar

The changes have also been backported to branch-2-6.
